src/gwt/acesupport/acemode/r_code_model.js

```diff
@@ -37,6 +37,15 @@ function isOneOf(object, array)
    return false;
 }
 
+function isControlFlowFunctionKeyword(value)
+{
+   return value === "if" ||
+          value === "for" ||
+          value === "while" ||
+          value === "repeat" ||
+          value === "function";
```

Comment:
Should `else` be on this list?

Reply:
`else` needs its own special handling, which I haven't touched in this PR. For example, in this case, normally one wants indentation at the first column:

```
if (foo)
  if (bar)
    x <- 1
|
```

but if `else` is inserted, then we need to outdent:

```
if (foo)
  if (bar)
    x <- 1
  else
```